inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_greedy_decoder.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <functional_test_utils/ov_tensor_utils.hpp>
+#include <ngraph_functions/builders.hpp>
+#include <shared_test_classes/single_layer/ctc_greedy_decoder.hpp>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "test_utils/cpu_test_utils.hpp"
+
+using namespace InferenceEngine;
+using namespace CPUTestUtils;
+using namespace ov::test;
+
+namespace CPULayerTestsDefinitions {
+
+using TargetShapeParams = std::tuple<size_t,   // Sequence length T
+                                     size_t,   // Batch size N
+                                     size_t>;  // Number of classes
+
+using InputShapeParams = std::pair<std::vector<ov::PartialShape>,    // bounds for input dynamic shape
```

Comment:
maybe better to pass PartialShape for `T` `N` `C` not for each inputs?

Reply:
Hi yes, currently I just pass `T`, `N`, `C` as the `TargetShapeParams` for constructing `data` & `seq_len` shapes, I don't pass input shapes for `data` and `seq_len` separately. Maybe the naming `TargetShapeParams` is a bit misleading and I will change it to proper one. Thanks